src/training/train.py

```diff
@@ -177,21 +199,22 @@ def evaluate(model, data, epoch, args, tb_writer=None):
         all_image_features, all_text_features = [], []
         with torch.no_grad():
             for i, batch in enumerate(dataloader):
-                images, texts = batch
-                images = images.to(device=device, dtype=cast_dtype, non_blocking=True)
-                texts = texts.to(device=device, non_blocking=True)
+                batch = Batch(**batch).to(device=device, non_blocking=True, dtypes={'image': cast_dtype})
 
                 with autocast():
-                    image_features, text_features, logit_scale = model(images, texts)
+                    output = model(**batch)
                     # features are accumulated in CPU tensors, otherwise GPU memory exhausted quickly
                     # however, system RAM is easily exceeded and compute time becomes problematic
+                    image_features, text_features, logit_scale = output['image_features'], \
+                                                                 output['text_features'], \
+                                                                 output['logit_scale']
                     all_image_features.append(image_features.cpu())
                     all_text_features.append(text_features.cpu())
                     logit_scale = logit_scale.mean()
                     logits_per_image = logit_scale * image_features @ text_features.t()
                     logits_per_text = logits_per_image.t()
 
-                    batch_size = images.shape[0]
+                    batch_size = batch.size()
                     labels = torch.arange(batch_size, device=device).long()
                     total_loss = (
```

Comment:
Maybe I am missing something here. Where are you adding rest of the losses to the total loss that are returned from FLAVA?

Reply:
or do you plan to make this particular part more generic by allowing loss registry? How are you pretraining FLAVA currently?